specs/source.go

```diff
@@ -57,6 +60,9 @@ func (s *Source) SetDefaults() {
 	if s.Concurrency == 0 {
 		s.Concurrency = defaultConcurrency
 	}
+	if s.Scheduler == "" {
+		s.Scheduler = "dfs"
```

Comment:
move this check to the `plugins/source/plugin.go`, see comment there

Reply:
Disagree, we should set a default if empty, but tell the user if they entered something invalid.